openmc/mgxs_library.py

```diff
@@ -2099,6 +2429,107 @@ def get_by_name(self, name):
+            domain is subdivided into. This is required when
+            :param:`target_representation` is "angle".
+        num_azimuthal : int, optional
+            Number of equal width angular bins that the azimuthal angular
+            domain is subdivided into. This is required when
+            :param:`target_representation` is "angle".
+
+        Returns
+        -------
+        openmc.MGXSLibrary
+            Multi-group Library with the same data as self, but represented as
+            specified in :param:`target_representation`.
+
+        """
+
+        check_value('target_representation', target_representation,
+                    _REPRESENTATIONS)
+        if target_representation == 'angle':
+            check_type('num_polar', num_polar, Integral)
+            check_type('num_azimuthal', num_azimuthal, Integral)
+            check_greater_than('num_polar', num_polar, 0)
+            check_greater_than('num_azimuthal', num_azimuthal, 0)
+
+        library = copy.deepcopy(self)
+        for i, xsdata in enumerate(self.xsdatas):
+            library.xsdatas[i] = \
+                xsdata.convert_representation(target_representation,
+                                              num_polar, num_azimuthal)
+        return library
+
+    def convert_scatter_format(self, target_format, target_order):
+        """Produce a new MGXSLibrary object with the same data, but converted
+        to the new scatter format and order
+
+        Parameters
+        ----------
+        target_format : {'tabular', 'legendre', 'histogram'}
+            Representation of the scattering angle distribution
+        target_order : int
+            Either the Legendre target_order, number of bins, or number of
+            points used to describe the angular distribution associated with
+            each group-to-group transfer probability
+
+        Returns
+        -------
+        openmc.MGXSLibrary
+            Multi-group Library with the same data as self, but with the
+            scatter format represented as specified in :param:`target_format`
+            and :param:`target_order`.
+
+        """
+
+        check_value('target_format', target_format, _SCATTER_TYPES)
+        check_type('target_order', target_order, Integral)
+        if target_format == 'legendre':
+            check_greater_than('target_order', target_order, 0, equality=True)
+        else:
+            check_greater_than('target_order', target_order, 0)
+
+        library = copy.deepcopy(self)
+        for i, xsdata in enumerate(self.xsdatas):
+            library.xsdatas[i] = \
+                xsdata.convert_scatter_format(target_format, target_order)
+
+        return library
+
+    def convert_to_continuous_energy(self, h5_filename='ce_mgxs.h5',
+                                     library_filename='ce_mgxs.xml'):
+        """Converts the MGXSLibrary object to an equivalent
+        library of openmc.data.IncidentNeutron objects
+
```

Comment:
If you are going to introduce this, perhaps add a more detailed explanatory paragraph to the docstring detailing how this can be supplied to a CE MC simulation.